File: thesis/src/sources/echo-function/context_0_0.py
import values
import sys
value_file = open("t8QUHe5mW_values.txt", "w")
exception_file = open("t8QUHe5mW_exceptions.txt", "w")
def write_separator():
    value_file.write("--t8QUHe5mW-- SEP")
    exception_file.write("--t8QUHe5mW-- SEP")
    sys.stderr.write("--t8QUHe5mW-- SEP")
    sys.stdout.write("--t8QUHe5mW-- SEP")
    sys.stdout.flush()
    sys.stderr.flush()
    value_file.flush()
    exception_file.flush()
def send_value(value):
    values.send_value(value_file, value)
def send_exception(exception):
    values.send_exception(exception_file, exception)
def send_specific_value(value):
    values.send_evaluated(value_file, value)
def send_specific_exception(exception):
    values.send_evaluated(exception_file, exception)
try:
    write_separator()
    from submission import *
except Exception as e:
    raise e
write_separator()
try:
    send_value(echo('input-1'))
except Exception as e:
    send_exception(e)
else:
    send_exception(None)
write_separator()
try:
    send_value(echo('input-2'))
except Exception as e:
    send_exception(e)
else:
    send_exception(None)
value_file.close()
exception_file.close()

File: thesis/src/sources/echo-function/Context00.hs
{-# LANGUAGE NamedFieldPuns #-}
module Context00 where
import System.IO (hPutStr, stderr, stdout, hFlush)
import System.Environment
import qualified Values
import Control.Monad.Trans.Class
import Control.Exception
import EvaluationUtils
import qualified Submission
value_file = "yq5AlxCOr_values.txt"
exception_file = "yq5AlxCOr_exceptions.txt"
writeSeparator :: IO ()
writeSeparator = do
    hPutStr stderr "--yq5AlxCOr-- SEP"
    hPutStr stdout "--yq5AlxCOr-- SEP"
    appendFile value_file "--yq5AlxCOr-- SEP"
    appendFile exception_file "--yq5AlxCOr-- SEP"
    hFlush stdout
    hFlush stderr
sendValue :: Values.Typeable a => a -> IO ()
sendValue = Values.sendValue value_file
sendException :: Exception e => Maybe e -> IO ()
sendException = Values.sendException exception_file
sendSpecificValue :: EvaluationResult -> IO ()
sendSpecificValue = Values.sendEvaluated value_file
sendSpecificException :: EvaluationResult -> IO ()
sendSpecificException = Values.sendEvaluated exception_file
handleException :: Exception e => (Either e a) -> Maybe e
handleException (Left e) = Just e
handleException (Right _) = Nothing
main = do
    writeSeparator
    writeSeparator
    result0 <- catch
        (return (Submission.echo ("input-1"))
            >>= \r -> sendValue (r)
            >> let ee = (Nothing :: Maybe SomeException) in sendException (ee))
        (\e -> let ee = (Just (e :: SomeException)) in sendException (ee))
    writeSeparator
    result1 <- catch
        (return (Submission.echo ("input-2"))
            >>= \r -> sendValue (r)
            >> let ee = (Nothing :: Maybe SomeException) in sendException (ee))
        (\e -> let ee = (Just (e :: SomeException)) in sendException (ee))
    putStr ""